From 7876d1a2efa61a5aea066db5c3659b913c0e127d Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 12 Jan 2024 14:32:14 +0100
Subject: [PATCH 0573/1695] dts: stm32: share pmic regulators for non secure
 world on stm32mp2

Share PMIC regulators for non secure world.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e1484d088772c9f3080f27ae0eff9d51561a16ca dts: stm32: share pmic regulators for non secure world on stm32mp2]
Change-Id: I536535b2890f514753d4eef1e27bfd5449de0ecf
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/350984
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  |  4 ++++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 14 +++++++++++---
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index b8ee298d2..c709304df 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -374,6 +374,10 @@
 };
 
 &scmi_regu {
+	voltd-vddcore {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
+		voltd-supply = <&vddcore>;
+	};
 	voltd-v1v8 {
 		reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
 		voltd-supply = <&v1v8>;
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 74922e5a7..663c03cb5 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -408,15 +408,23 @@
 };
 
 &scmi_regu {
-	scmi_v3v3: voltd-v3v3 {
+	voltd-vddcore {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
+		voltd-supply = <&vddcore>;
+	};
+	voltd-v1v8 {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
+		voltd-supply = <&v1v8>;
+	};
+	voltd-v3v3 {
 		reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
 		voltd-supply = <&v3v3>;
 	};
-	scmi_vdd_emmc: voltd-vdd-emmc {
+	voltd-vdd-emmc {
 		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
 		voltd-supply = <&vdd_emmc>;
 	};
-	scmi_vdd_sdcard: voltd-vdd_sdcard {
+	voltd-vdd-sdcard {
 		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
 		voltd-supply = <&vdd_sdcard>;
 	};
-- 
2.47.2

